### PR TITLE
343026 add tests for using project-location variables

### PR DESCRIPTION
**Pull request by kwin**
see https://bugs.eclipse.org/bugs/show_bug.cgi?id=343026

### CODE REVIEW

File: org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/launch/LaunchingUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Sonatype, Inc.
```
Comment (fbricon) [inline org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/launch/LaunchingUtilsTest.java]: Copyright is yours (or your employers, if you contributed during work time), not Sonatype

File: org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/launch/LaunchingUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *      Sonatype, Inc. - initial API and implementation
```
Comment (fbricon) [inline org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/launch/LaunchingUtilsTest.java]: Contributors:
     Konrad Windszus (Employer?) - initial API and implementation

File: org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/launch/LaunchingUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    Assert.assertEquals(project.getLocation().toOSString(), newProjectLocation);
+  }
+
+  /**
+   * Similar to {@link #createExisting}
+   * 
+   * @throws CoreException
+   * @throws IOException
+   */
+  protected void moveExistingProjectLocation(IProject project, File projectSourceFolder, File projectTargetFolder)
```
Comment (fbricon) [inline org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/launch/LaunchingUtilsTest.java]: there's a IProject.move : method http://help.eclipse.org/mars/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Freference%2Fapi%2Forg%2Feclipse%2Fcore%2Fresources%2FIProject.html

Comment (kwin) [inline org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/launch/LaunchingUtilsTest.java]: That does not work if you want to keep the project's name but only move the underlying location. Therefore I had to implement my own method.

File: org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/launch/LaunchingUtilsTest.java (hunk 1 above)
Comment (kwin) [inline org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/launch/LaunchingUtilsTest.java]: That license header was automatically generated by Eclipse. I will modify accordingly.

**Comment by fbricon** [comment]
Merged and rebased as https://github.com/tesla/m2e-core-tests/commit/40a09827c035a5404ae048f95ab47af9b20aadff.

Thanks @kwin

**Pull request closed without merging**